x-pack/plugins/security_solution/scripts/endpoint/resolver_generator_script.ts

```diff
@@ -4,14 +4,40 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 /* eslint-disable no-console */
-import * as path from 'path';
 import yargs from 'yargs';
-import * as url from 'url';
-import fetch from 'node-fetch';
 import { Client, ClientOptions } from '@elastic/elasticsearch';
 import { ResponseError } from '@elastic/elasticsearch/lib/errors';
+import { KbnClient, ToolingLog } from '@kbn/dev-utils';
+import { AxiosResponse } from 'axios';
+import { KibanaConfig } from '@kbn/dev-utils/target/kbn_client/kbn_client_requester';
+import fetch, { RequestInit as FetchRequestInit } from 'node-fetch';
 import { indexHostsAndAlerts } from '../../common/endpoint/index_data';
 import { ANCESTRY_LIMIT } from '../../common/endpoint/generate_data';
+import { FLEET_SETUP_API_ROUTES, SETUP_API_ROUTE } from '../../../ingest_manager/common/constants';
+import {
+  CreateFleetSetupResponse,
+  PostIngestSetupResponse,
+} from '../../../ingest_manager/common/types/rest_spec';
+
+class KbnClientWithApiKeySupport extends KbnClient {
+  private kibanaUrlNoAuth: string;
+  constructor(log: ToolingLog, kibanaConfig: KibanaConfig) {
+    super(log, kibanaConfig);
+    const kibanaUrl = this.resolveUrl(kibanaConfig.url);
+    const matches = kibanaUrl.match(/(https?:\/\/)(.*\:.*\@)(.*)/);
+    // strip auth from url
+    this.kibanaUrlNoAuth =
+      matches && matches.length >= 3
+        ? matches[1] + matches[3].replace('/', '')
+        : kibanaUrl.replace('/', '');
+  }
+  requestWithApiKey(path: string, init?: RequestInit | undefined): Promise<Response> {
```

Comment:
Can you add a comment with a quick overview of what this function is used for? It looks to me like it's simulating a request from an Elastic Agent to Kibana that wants to enroll in fleet, is that right?